pkg/apis/core/validation/utils_test.go

```diff
@@ -91,4 +93,105 @@ var _ = Describe("Utils tests", func() {
+				errorList := ValidateCapabilitiesDefinition(capabilities, dummyPath)
+				Expect(errorList).To(ConsistOf(expectedError))
+			},
+			Entry("reject empty capability name", core.Capabilities{"architecture": "amd64", "": "gen1,gen2,gen3"}, []gomegatypes.GomegaMatcher{
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeInvalid),
+					"Field": Equal(dummyPath.String()),
+				})),
+			}),
+			Entry("empty capability values", core.Capabilities{"architecture": "amd64", "hypervisorType": ""}, []gomegatypes.GomegaMatcher{
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeRequired),
+					"Field": Equal(dummyPath.Child("hypervisorType").String()),
+				})),
+			}),
+			Entry("missing architecture capability", core.Capabilities{"hypervisorType": "gen1,gen2,gen3"}, []gomegatypes.GomegaMatcher{
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeRequired),
+					"Field": Equal(dummyPath.Child("architecture").String()),
+				})),
+			}),
+		)
+
+		It("should unmarshal correct CapabilitiesSet", func() {
+			var CapabilitiesSet = []v1.JSON{
+				{Raw: []byte(`{"architecture":"amd64","hypervisorType":"gen2"}`)},
+				{Raw: []byte(`{"architecture":"amd64","hypervisorType":"gen2"}`)},
+				{Raw: []byte(`{"architecture":"arm64","hypervisorType":"gen1"}`)},
+				{Raw: []byte(`{"architecture":"arm64","hypervisorType":"gen2,gen3"}`)},
+			}
+			capabilitiesSet, err := UnmarshalCapabilitiesSet(CapabilitiesSet, dummyPath)
+
+			Expect(err).To(BeNil())
+			Expect(capabilitiesSet).To(HaveLen(4))
+			Expect(capabilitiesSet[0]).To(Equal(core.Capabilities{"architecture": "amd64", "hypervisorType": "gen2"}))
+		})
+
+		It("should error on unparsable CapabilitiesSet", func() {
+			var CapabilitiesSet = []v1.JSON{
+				{Raw: []byte(`{"architecture":"amd64","hypervisorType":"gen2"}`)},
+				{Raw: []byte(`{"architecture":"amd64","hype....🆘`)},
+			}
+
+			capabilitiesSet, err := UnmarshalCapabilitiesSet(CapabilitiesSet, dummyPath)
+
+			Expect(err).To(ConsistOf([]gomegatypes.GomegaMatcher{
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeInvalid),
+					"Field": Equal("dummy[1]"),
+				})),
+			}))
+			Expect(capabilitiesSet[1]).To(BeEmpty())
+		})
+
+		It("should sanitize capability values on parsing", func() {
+			var capabilities2 core.Capabilities = map[string]string{
+				"architecture":   "  amd64 ,arm64 ,amd32, , 'asdas   '    , I look weird  ",
+				"hypervisorType": `"gen1", "gen4"`,
+			}
+			parsedCapabilities := ParseCapabilitiesValues(capabilities2)
+			architectureSet := parsedCapabilities["architecture"]
+			hypervisorTypeSet := parsedCapabilities["hypervisorType"]
+
+			Expect(architectureSet.Contains("amd64", "arm64", "amd32", "I look weird", "asdas")).To(BeTrue())
+			Expect(architectureSet).To(HaveLen(5))
+			Expect(hypervisorTypeSet.Contains("gen1", "gen4")).To(BeTrue())
```

Comment:
Why do we use `.Contains` here instead of `.Equals`? Doesn't the order have relevance with regards to importance of the values?